### ISSUE TITLE
maxDuration always leads to the route returning 404 in Remix

### ISSUE THREAD

**Issue by Klingefjord** [zeit/now#10590]
See https://github.com/remix-run/remix/issues/7537

**Comment by mint-pz** [comment]
I got the 404 route error with Nextjs project recently without any changing in nextjs or vercel configs when deploying to vercel (the error doesn't happen in localhost)

<img width="1037" alt="image" src="https://github.com/vercel/vercel/assets/98034637/96639534-1334-494c-b16e-1c3e3212f26f">

**Comment by poksme** [comment]
I do not know if this can help people in this thread, but I spent the last couple of days debugging why 
```js
export const config = { maxDuration: 42 };
```
was ignored by vercel in my remix (1.18) project.

~I ended up finding a big _gotcha_ : in the `remix.config.js`, the simple fact of explicitly targetting vercel using `serverBuildTarget: "vercel"` completely striped the config.~

It was actually the vercel CLI version, I was lock on an older version.

Creating a minimal project to reproduce the problem and slowly adding similar configuration helped me. + fixing all warnings, namely the locked VERCEL_CLI warning.
